lib/elixir/lib/kernel.ex

```diff
@@ -3029,16 +3029,23 @@ defmodule Kernel do
     in_module? = (__CALLER__.context == nil)
 
     right = case bootstrapped?(Macro) do
-      true  -> Macro.expand(right, __CALLER__)
+      true  -> Macro.prewalk(right, &Macro.expand(&1, __CALLER__))
```

Comment:
Ok. So to be clear, we need to remove this and call expand whenever we find a `{:|, ..., ...}` inside `in_list` and `in_module?` is false. If `in_module?` is true then we don't care because we have already extracted it to a var.